code/parachain/runtime/picasso/src/lib.rs

```diff
@@ -178,6 +196,78 @@ parameter_types! {
 	pub const SS58Prefix: u8 = 49;
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Default)]
+pub struct Router {}
+
+impl ModuleRouter for Router {
+	fn get_route_mut(&mut self, _module_id: &ModuleId) -> Option<&mut dyn Module> {
+		None
+	}
+
+	fn has_route(_module_id: &ModuleId) -> bool {
+		false
+	}
+
+	fn lookup_module_by_port(_port_id: &PortId) -> Option<ModuleId> {
+		None
+	}
+}
+
+pub struct IbcDenomToAssetIdConversion;
+
+impl DenomToAssetId<Runtime> for IbcDenomToAssetIdConversion {
+	type Error = DispatchError;
+
+	fn from_denom_to_asset_id(denom: &String) -> Result<CurrencyId, Self::Error> {
+		let denom_bytes = denom.as_bytes().to_vec();
+		if let Some(id) = IbcDenoms::<Runtime>::get(&denom_bytes) {
+			return Ok(id)
+		}
+
+		let asset_id =
+			<currency_factory::Pallet<Runtime> as CurrencyFactoryT>::create(RangeId::IBC_ASSETS)?;
```

Comment:
@benluelo this will be affected by assets rewrite

Reply:
CC @PoisonPhang

Reply:
Yes, this process will changes when done through assets registry.
Currently planning on using the `denom_bytes` as a type of foreign asset to register and create an asset ID through assets registry.